### PR TITLE
python: fix spack external find for @3.13:

### PR DESCRIPTION
**Pull request by tukss**
`spack external find python` does not work for me with Python 3.13.0.

The reason is that Python >= 3.13 does not have the crypt variant anymore. Still no matter if the test for crypt succeeds (which it can on Fedora providing [its own crypt module](https://github.com/fedora-python/crypt_r) for Python 3.13) or fails, it will add `+crypt` or `~crypt`, which both fail because the variant only exists until Python 3.12.

The code for checking the version looks a bit awkward. I'm grateful for suggestions how to make this cleaner.

This is the output of `spack -d external find python` with Python 3.13.0 before the change.
```
==> [2024-12-17-11:08:14.103830] Parsing failed [spec_str="python@3.13.0 +pythoncmd+readline+sqlite3+dbm+zlib+bz2+lzma+ctypes+tkinter+uuid+ssl+pyexpat~tix~crypt", error=Invalid variant crypt for spec python@3.13.0+bz2~crypt+ctypes+dbm+lzma+pyexpat+pythoncmd+readline+sqlite3+ssl~tix+tkinter+uuid+zlib.
crypt is only available for python when satisfying one of (@:3.12 arch=linux-None-None, @:3.12 arch=darwin-None-None).]
==> [2024-12-17-11:08:14.103979] The following files in /usr/bin were decidedly not part of the package python: /usr/bin/python, /usr/bin/python3
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/python/package.py [hunk 1]
```diff
@@ -364,7 +365,9 @@ def determine_variants(cls, exes, version_str):
                 variants += "~tix"
 
         # Some modules are platform-dependent
-        if sys.platform != "win32":
+        python_version = spack.version.StandardVersion.from_string(version_str)
+        python313_version = spack.version.StandardVersion.from_string("3.13")
+        if sys.platform != "win32" and python_version < python313_version:
```
Comment (rbberger) [inline var/spack/repos/builtin/packages/python/package.py]:
```suggestion
        if sys.platform != "win32" and Version(version_str) < Version("3.13"):
```

Comment (tukss) [reply var/spack/repos/builtin/packages/python/package.py]: Perfect. I was looking for that. I'll also fix the imports.

**Review by rbberger — APPROVED** [review]
LGTM

**Comment by tukss** [comment]
@spackbot help

**Comment by tukss** [comment]
@spackbot re-run pipeline

**Comment by tukss** [comment]
@spackbot re-run pipeline

**Comment by tukss** [comment]
Looks like the pipeline is hitting some timeouts. Trying again.

**Comment by tukss** [comment]
Thanks for the update. Looks like it's passing. Ready to merge from my side.

**Review by scheibelp — APPROVED** [review]

**Pull request merged**